packages/fetchai/skills/aries_alice/handlers.py

```diff
@@ -0,0 +1,236 @@
+            sender=self.context.agent_address,
+            protocol_id=HTTP_PROTOCOL_PUBLIC_ID,
+            context=context,
+            message=HttpSerializer().encode(request_http_message),
+        )
+        self.context.outbox.put(envelope)
+
+    def setup(self) -> None:
+        """
+        Implement the setup.
+
+        :return: None
+        """
+        pass
+
+    def handle(self, message: Message) -> None:
+        """
+        Implement the reaction to an envelope.
+
+        :param message: the message
+        :return: None
+        """
+        message = cast(DefaultMessage, message)
+        if message.performative == DefaultMessage.Performative.BYTES:
+            content_bytes = message.content
+            content = json.loads(content_bytes)
+            self.context.logger.info("Received message content:" + str(content))
+            if "@type" in content:
+                # self.handle_received_invite(content)
+                self._admin_post("/connections/receive-invitation", content)
+
+    def teardown(self) -> None:
+        """
+        Implement the handler teardown.
+
+        :return: None
+        """
+        pass
+
+    def handle_received_invite(self, invite_detail: Dict):
+        for details in invite_detail:
+            try:
+                url = urlparse(details)
+                query = url.query
+                if query and "c_i=" in query:
+                    pos = query.index("c_i=") + 4
+                    b64_invite = query[pos:]
+                else:
+                    b64_invite = details
+            except ValueError:
+                b64_invite = details
+
+            if b64_invite:
+                try:
+                    padlen = 4 - len(b64_invite) % 4
+                    if padlen <= 2:
+                        b64_invite += "=" * padlen
+                    invite_json = base64.urlsafe_b64decode(b64_invite)
+                    details = invite_json.decode("utf-8")
+                except binascii.Error:
+                    pass
+                except UnicodeDecodeError:
+                    pass
+
+            if details:
+                try:
+                    json.loads(details)
+                    break
+                except json.JSONDecodeError as e:
+                    self.context.logger.error("Invalid invitation:", str(e))
+
+        self._admin_post("/connections/receive-invitation", details)
+
+
+class AriesDemoHttpHandler(Handler):
+    """This class represents alice's handler for HTTP messages."""
+
+    SUPPORTED_PROTOCOL = HttpMessage.protocol_id  # type: Optional[ProtocolId]
+
+    def __init__(self, **kwargs):
+        """Initialize the handler."""
+        self.admin_host = kwargs.pop("admin_host", DEFAULT_ADMIN_HOST)
+        self.admin_port = kwargs.pop("admin_port", DEFAULT_ADMIN_PORT)
+
+        super().__init__(**kwargs)
+
+        self.admin_url = "http://{}:{}".format(self.admin_host, self.admin_port)
+        self.connection_id = ""
```

Comment:
I'd initialise as follows: `self.connection_id = None  # type: Optional[str]` and then use the fact that `None` means it is not initialised in the control flow. O/w you have to handle the corner case where the received connection_id is `""`

Reply:
Done